### PR TITLE
Interactivity API: Fix null and number strings as namespaces runtime error.

### PR DESCRIPTION
**Pull request by cbravobernal**
## What?

Right now, creating a namespace called `null` or an integer `24` will kill the runtime.

## Why?

We are doing a JSON.parse() function that returns `null` type or `number` type if receives `null` or `number` strings.

## How?

By checking if it is an object.

## Testing Instructions

1) Create an interactive block.
2) Use null or a number for the namespace in render.php
3) The interactive block should work.

### CODE REVIEW

File: packages/e2e-tests/plugins/interactive-blocks/namespace/render.php [hunk 1]
```diff
@@ -18,6 +18,14 @@
 	<a data-wp-bind--href="state.url" data-testid="object namespace"></a>
 </div>
 
+<div data-wp-interactive="null">
+	<a data-wp-bind--href="state.url" data-testid="null namespace"></a>
+</div>
+
+<div data-wp-interactive="2">
+	<a data-wp-bind--href="state.url" data-testid="number namespace"></a>
+</div>
```
Comment (sirreal) [inline packages/e2e-tests/plugins/interactive-blocks/namespace/render.php:27]: These would've had similar interesting behavior, all the JSON values:

```html
<div data-wp-interactive="true"></div>
<div data-wp-interactive="false"></div>
<div data-wp-interactive="[]"></div>
<div data-wp-interactive='"my namespace is a quoted string"'></div>
```

**Review by sirreal — COMMENTED** [review]
I checked the server side, it seems to handle this well (decodes to associative array and checks for is_array):

https://github.com/WordPress/wordpress-develop/blob/581f8d5eebcc57c8f2e8fe05e44db6cc183e1c26/src/wp-includes/interactivity-api/class-wp-interactivity-api.php#L558-L565

File: packages/interactivity/src/vdom.ts [hunk 2]
```diff
@@ -100,7 +100,8 @@ export function toVdom( root: Node ): Array< ComponentChild > {
 					const namespace = regexResult?.[ 1 ] ?? null;
 					let value: any = regexResult?.[ 2 ] ?? attributeValue;
 					try {
-						value = value && JSON.parse( value );
+						const parsedValue = JSON.parse( value );
+						value = isObject( parsedValue ) ? parsedValue : value;
 					} catch ( e ) {}
```
Comment (sirreal) [inline packages/interactivity/src/vdom.ts]: Nit: we can omit `( e )` when not using the error:
```suggestion
					} catch {}
```

Comment (cbravobernal) [reply packages/interactivity/src/vdom.ts]: Fixed in https://github.com/WordPress/gutenberg/pull/61960/commits/b7b6da8f447a974daf807c8034119ccd6824fcaa

File: packages/e2e-tests/plugins/interactive-blocks/namespace/view.js [hunk 3]
```diff
@@ -14,3 +21,50 @@ store( 'other', {
 		url: '/other-store-url',
 	},
 } );
+
+store( 'null', {
+	state: {
+		url: '/some-url',
```
Comment (sirreal) [inline packages/e2e-tests/plugins/interactive-blocks/namespace/view.js]: I'd like to see different values in the different stores. I think it's unlikely, but one could imagine falsy values evaluating to the same store.

**Review by sirreal — APPROVED** [review]
I tested this and it does allow me to make a (terribly named) store with the namespace `null`.

File: packages/e2e-tests/plugins/interactive-blocks/namespace/view.js (hunk 3 above)
Comment (cbravobernal) [reply packages/e2e-tests/plugins/interactive-blocks/namespace/view.js:27]: TBH, I'm not 100% sure if we should allow this values to be namespaces 😅

**Comment by cbravobernal** [comment]
> I tested this and it does allow me to make a (terribly named) store with the namespace `null`.

TBH, I'm not 100% sure if we should allow this values to be namespaces 😅

File: packages/e2e-tests/plugins/interactive-blocks/namespace/view.js (hunk 3 above)
Comment (cbravobernal) [reply packages/e2e-tests/plugins/interactive-blocks/namespace/view.js]: Done in https://github.com/WordPress/gutenberg/pull/61960/commits/64e275d4fdd6ff47e30a45e5933a0dba5a92f892

**Pull request merged**